contracts/src/components/movable.cairo

```diff
@@ -16,3 +19,61 @@ struct ArrivalTime {
     entity_id: u128,
     arrives_at: u64,
 }
+
+#[derive(Component, Copy, Drop, Serde, SerdeLen)]
+struct Travel {
+    #[key]
+    entity_id: u128,
+    arrival_time: u64,
+    departure_time: u64,
+    departure_coord_x: u32,
+    departure_coord_y: u32,
+    // last stop means last location before reaching destination
+    // e.g if a caravan is traveling from A -> B -> A,
+    // last_stop_coord will be that of B.
+    last_stop_coord_x: u32,
+    last_stop_coord_y: u32
+}
+
+
+#[generate_trait]
+impl TravelImpl of TravelTrait {
+    fn live_location(self: @Travel, destination: Coord, timestamp: u64) -> Coord {
```

Comment:
that's a great idea to implement this trait, think this should be implemented for movable though